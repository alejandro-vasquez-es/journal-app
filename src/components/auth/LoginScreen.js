import React from 'react';
import { useDispatch } from 'react-redux';

import { Link } from 'react-router-dom'
import { login, startLoginEmailPassword, startGoogleLogin } from '../../actions/auth';
import { useForm } from '../../hooks/useForms';

export const LoginScreen = () => {

	const dispatch = useDispatch();

	const [formValues, handleInputChange] = useForm({
		email: 'correo@gmail.com',
		password: 'test'
	});

	const { email, password } = formValues;

	const handleLogin = (e) => {
		e.preventDefault();
		dispatch(startLoginEmailPassword(email, password))
	}

	const handleGoogleLogin = () => {
		dispatch(startGoogleLogin());
	}

	return (
		<>
			<h2 className="auth__title">Login</h2>

			<form
				className="auth__form"
				onSubmit={handleLogin}
			>

				<input
					type="text"
					placeholder="email"
					name="email"
					className="auth__input"
					autoComplete='off'
					value={email}
					onChange={handleInputChange}
				/>

				<input
					type="password"
					placeholder="Password"
					name="password"
					className="auth__input"
					value={password}
					onChange={handleInputChange}
				/>

				<button
					className="button"
					type="submit"
				>
					Login
				</button>

				<div className="google__container">

					<p><b>Login with social networks </b></p>

					<div
						className="google-btn"
						onClick={handleGoogleLogin}
					>
						<div className="google-icon-wrapper">
							<img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
						</div>
						<p className="btn-text">
							<b>Sign in with google</b>
						</p>
					</div>

				</div>

				<Link
					className='link'
					to="/auth/register"
				>
					Create new account
				</Link>

			</form>

		</>
	)
}
