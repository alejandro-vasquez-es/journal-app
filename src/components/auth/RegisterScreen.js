import React from 'react'
import { Link } from 'react-router-dom'
import { useForm } from '../../hooks/useForms'

export const RegisterScreen = () => {

	// name: 'Alejandro',
	// email: 'correo@correo.com',
	// password: '123456',
	// password2: '123456' 

	const [formValues, handleInputChange] = useForm({
		name: 'Alejandro',
		email: 'correo@correo.com',
		password: '123456',
		password2: '123456'

	});

	const { name, email, password, password2 } = formValues;

	const handleLogin = (e) => {
		e.preventDefault();
		console.log(formValues);
	}

	const isFormValid = () => {
		//TODO:
	}

	return (
		<>
			<h2 className="auth__title">Register</h2>

			<form className="auth__form">

				<input
					type="text"
					placeholder="Name"
					name="name"
					className="auth__input"
					autoComplete='off'
					value={name}
					onChange={handleInputChange}
				/>

				<input
					type="text"
					placeholder="email"
					name="email"
					className="auth__input"
					autoComplete='off'
					value={email}
					onChange={handleInputChange}
				/>

				<input
					type="password"
					placeholder="Password"
					name="password"
					className="auth__input"
					value={password}
					onChange={handleInputChange}
				/>

				<input
					type="password"
					placeholder="Confirm password"
					name="password2"
					className="auth__input"
					value={password2}
					onChange={handleInputChange}
				/>

				<button
					className="button mb-5"
					type="submit"
					onClick={handleLogin}
				>
					Register
				</button>

				<Link
					className='link mt-1'
					to="/auth/login"
				>
					Already registered?
				</Link>

			</form>

		</>
	)
}
