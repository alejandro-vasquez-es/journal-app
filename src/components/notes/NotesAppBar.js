import React from 'react'

export const NotesAppBar = () => {
	return (
		<div className="notes__appbar">
			<span> 16 de febrero del 2022 </span>

			<div className='notes__btn-container'>
				<button className='btn'>
					Picture
				</button>
				<button className='btn'>
					Save
				</button>
			</div>
		</div>
	)
}
