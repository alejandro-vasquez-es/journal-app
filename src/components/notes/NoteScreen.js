import React from 'react'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
	return (
		<div className="notes__main-content">

			<NotesAppBar />

			<div className='notes__content'>

				<input
					type="text"
					placeholder="jijijajaja"
					className="notes__title-input"
				/>

				<textarea
					placeholder='what happened today'
					className='notes__textarea'
				>

				</textarea>

				<div className='notes__image'>
					<img
						alt='landcape'
						src='https://images.pexels.com/photos/592077/pexels-photo-592077.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500'
					/>
				</div>

			</div>

		</div>
	)
}
