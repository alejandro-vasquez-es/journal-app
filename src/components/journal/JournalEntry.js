import React from 'react';
import moment from 'moment'
import { useDispatch } from 'react-redux';
import { activeNote } from '../../actions/notes';

export const JournalEntry = ({ id, date, title, body, url }) => {

	const dispatch = useDispatch();
	const noteDate = moment(date);

	const goToActiveNote = (duration) => {
		const targetPosition = Math.max(document.documentElement.clientHeight || 0, window.innerHeight || 0);
		var startPosition = window.pageYOffset;
		var distance = targetPosition - startPosition;
		var startTime = null;

		function animation(currentTime) {
			if (startTime === null) startTime = currentTime;
			var timeElapsed = currentTime - startTime;
			var run = easeOutExpo(timeElapsed, startPosition, distance, duration);
			window.scroll(0, run);
			if (timeElapsed < duration) requestAnimationFrame(animation);
		}

		function easeOutExpo(t, b, c, d) {
			return c * (-Math.pow(2, -10 * t / d) + 1) + b - 16;
		}

		requestAnimationFrame(animation);
	}

	const handleEntryClick = () => {
		dispatch(activeNote(id, { date, title, body, url }));
		goToActiveNote(500);
	}

	return (
		<div
			className='journal__entry  animate__animated animate__slideInLeft animate__faster'
			onClick={handleEntryClick}
		>

			{
				url &&
				<div
					className='journal__entry-picture'
					style={{
						backgroundSize: 'cover',
						backgroundImage: `url(${url})`,
						backgroundPosition: 'center'
					}}
				>

				</div>
			}

			<div className='journal__entry-body'>
				<p className='journal__entry-title'>
					{title ? title : 'Untitled'}
				</p>
				<p className='journal__entry-content'>
					{body}
				</p>
			</div>

			<div className='journal__entry-date-box'>
				<p>{noteDate.format('dddd')}</p>
				<span>{noteDate.format('DD')}</span>
			</div>

		</div>
	)
}
