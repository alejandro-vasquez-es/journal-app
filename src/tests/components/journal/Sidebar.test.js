import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { Sidebar } from '../../../components/journal/Sidebar';
import { startLogout } from '../../../actions/auth'
import { startNewNote } from '../../../actions/notes';

jest.mock('../../../actions/auth', () => ({
	startLogout: jest.fn()
}));
jest.mock('../../../actions/notes', () => ({
	startNewNote: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
	auth: {
		uid: 'ABCDEF',
		name: 'TESTER'
	},
	ui: {
		loading: false,
		msgError: null
	},
	notes: {
		notes: [{
			id: '123',
			date: 1646876753881,
			title: "TITLE",
			url: "https://res.cloudinary.com/dj54vlrsd/image/upload/v1646877128/su5hfvcu8mkmwn48ca6w.jpg",
			body: 'BODY'
		}],
		active: null,
	}
};
let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
	<Provider store={store}>
		<MemoryRouter>
			<Sidebar />
		</MemoryRouter>
	</Provider>
);

describe('Tests in <Sidebar />', () => {

	test('should display propertly', () => {

		expect(wrapper).toMatchSnapshot();

	})

	test('should call the startLogout', () => {

		wrapper.find('button').simulate('click');

		expect(startLogout).toHaveBeenCalled();

	});

	test('should call the startNewNote', () => {

		wrapper.find('.journal__new-entry').simulate('click');

		expect(startNewNote).toHaveBeenCalled();

	})

})