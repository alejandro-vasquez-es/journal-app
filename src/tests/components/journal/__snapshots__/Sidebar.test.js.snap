// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Sidebar /> should display propertly 1`] = `
<aside
  className="journal__sidebar"
>
  <div
    className="journal__sidebar-navbar"
  >
    <h1
      className=" journal__sidebar-navbar-title mt-5"
    >
      <i
        className="far fa-moon"
      />
      <span>
          
        TESTER
      </span>
    </h1>
    <button
      className="btn"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
  <div
    className="journal__new-entry"
    onClick={[Function]}
  >
    <i
      className="far fa-calendar-plus fa-5x"
    />
    <p
      className="mt-5 mb-5"
    >
      New entry
    </p>
  </div>
  <div
    className="journal__entries"
  >
    <div
      className="journal__entry  animate__animated animate__slideInLeft animate__faster"
      onClick={[Function]}
    >
      <div
        className="journal__entry-picture"
        style={
          Object {
            "backgroundImage": "url(https://res.cloudinary.com/dj54vlrsd/image/upload/v1646877128/su5hfvcu8mkmwn48ca6w.jpg)",
            "backgroundPosition": "center",
            "backgroundSize": "cover",
          }
        }
      />
      <div
        className="journal__entry-body"
      >
        <p
          className="journal__entry-title"
        >
          TITLE
        </p>
        <p
          className="journal__entry-content"
        >
          BODY
        </p>
      </div>
      <div
        className="journal__entry-date-box"
      >
        <p>
          Wednesday
        </p>
        <span>
          09
        </span>
      </div>
    </div>
  </div>
</aside>
`;
